docs/site/Validation-REST-layer.md

```diff
@@ -175,3 +175,48 @@ of 10 digits separated by `-` after the 3rd and 6th digits.
 ```
 
 {% include tip.html content="RegExp can be converted into a string with `.source` to avoid escaping backslashes" %}
+
+## Customize validation errors
+
+Since the error is being caught at the REST layer, the best way to customize the
+errors is to
+[customize the sequence of actions](https://loopback.io/doc/en/lb4/Sequence.html#customizing-sequence-actions)
+using a
+[Provider](https://loopback.io/doc/en/lb4/apidocs.context.provider.html).
+
+To do so, create a Provider class in `src/providers` folder that implements
+`Provider<Reject>`. Using the
+[validation-app](https://github.com/strongloop/loopback-next/tree/master/examples/validation-app)
+as an example, here is how the Provider class look like:
+
+{% include code-caption.html content="/src/providers/my-validation-error-provider.ts" %}
+
+```ts
+import {Provider} from '@loopback/core';
+import {HandlerContext, HttpErrors, Reject} from '@loopback/rest';
+export class MyValidationErrorProvider implements Provider<Reject> {
+  //...
+  action({request, response}: HandlerContext, error: Error) {
+    // handle the error and send back the error response
+    // "response" is an Express Response object
+    response.setHeader('Content-Type', 'application/json');
+    const httpError = <HttpErrors.HttpError>error;
+    if (request.url === '/coffee-shops') {
+      // if this is a validation error
+      if (httpError.statusCode === 422) {
+        // you can customize the error message and status code here
+        const newError = {
+          message: 'My customized validation error message',
+          code: 'VALIDATION_FAILED',
+          resolution: 'Contact your admin for troubleshooting.',
+        };
+
+        // you can change the status code here too
+        response.status(422).send(JSON.stringify(newError));
+      }
+    }
+
+    response.status(httpError.statusCode).end(JSON.stringify(httpError));
```

Comment:
This isn't the best way how to present an error in the response. I believe this is going to drop the stack trace, which will make debugging difficult? At the same time, we don't want to print stack traces in production. All of this is handled by our default reject action. Can you please update this example to show how the custom error handler can fall back to the default implementation?

Reply:
@dhmlau , in our shopping example sequence, we catch an error like this : https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/sequence.ts#L72  inside the sequence, and pass them to default reject action .

Reply:
@emonddr, thanks for pointing to the code snippet!